packages/gatsby/src/redux/actions/public.js

```diff
@@ -364,9 +378,25 @@ ${reservedFields.map(f => `  * "${f}"`).join(`\n`)}
     internalComponentName = `Component${pascalCase(page.path)}`
   }
 
+  const truncateLongPaths = store.getState().config.truncateLongPaths
+  const pagePath = truncateLongPaths ? truncatePath(page.path) : page.path
+
+  if (pagePath.length > 255) {
```

Comment:
Is this the same for all OSes?

Reply:
Also I'm bit confused here - in here we check full path length, but in `truncatePath` function we do checks for path segments?

Reply:
1. I checked manually on Windows and MacOS. The limit is indeed 255 characters (not bytes, e.g. 255 Japanese characters (3 bytes) are still OK), which is what `.length` checks. Just in case, this also works for emoji -- for a path consisting of 2 byte (?) emojis the limit would be 127 emojis, but `.length` of the string is calculated as 254, so no problem.

2. Yes, that is a mistake. I'll fix it.

Reply:
I also checked manually on CentOS and it seems that it counts actual bytes, so an additional check is needed (MacOS / Windows vs others)